code/scripting/api/libs/ui.cpp

```diff
@@ -643,5 +673,177 @@ ADE_FUNC(getFictionMusicName, l_UserInterface_FictionViewer, nullptr,
 	return ade_set_args(L, "s", common_music_get_filename(SCORE_FICTION_VIEWER).c_str());
 }
 
+//**********SUBLIBRARY: UserInterface/TechRoom
+ADE_LIB_DERIV(l_UserInterface_TechRoom,
+	"TechRoom",
+	nullptr,
+	"API for accessing data related to the tech room UI.<br><b>Warning:</b> This is an internal "
+	"API for the new UI system. This should not be used by other code and may be removed in the future!",
+	l_UserInterface);
+
+ADE_FUNC(buildMissionList,
+	l_UserInterface_TechRoom,
+	nullptr,
+	"Builds the mission list for display. Must be called before the sim_mission handle will have data",
+	"number",
+	"Returns 1 when completed")
+{
+	API_Access = true;
```

Comment:
This is bad style. The ``API_Access`` variable is a global variable in your code. However, it's only used to modify the details of the code path of the function. Yes, this is done in some of the very old volition code as well, but it's bad there as well. Mostly cause it makes it a lot harder to follow where which values are set and what they will do.
Instead, you should pass it as a parameter to ``api_sim_room_build_mission_list`` and subfunctions

Reply:
Fixed